pkg/volume/util/types/types.go

```diff
@@ -36,22 +36,49 @@ type UniquePVCName types.UID
 type GeneratedOperations struct {
 	// Name of operation - could be used for resetting shared exponential backoff
 	OperationName     string
-	OperationFunc     func() (eventErr error, detailedErr error)
+	OperationFunc     func() (context OperationContext)
 	EventRecorderFunc func(*error)
-	CompleteFunc      func(*error)
+	CompleteFunc      func(CompleteFuncParam)
+}
+
+type OperationContext struct {
+	EventErr    error
+	DetailedErr error
+	Migrated    bool
+}
+
+func NewOperationContext(eventErr, detailedErr error, migrated bool) OperationContext {
+	return OperationContext{
+		EventErr:    eventErr,
+		DetailedErr: detailedErr,
+		Migrated:    migrated,
+	}
+}
+
+type CompleteFuncParam struct {
+	Err      *error
+	Migrated *bool
 }
 
 // Run executes the operations and its supporting functions
 func (o *GeneratedOperations) Run() (eventErr, detailedErr error) {
+	var context OperationContext
 	if o.CompleteFunc != nil {
-		defer o.CompleteFunc(&detailedErr)
+		c := CompleteFuncParam{
+			Err:      &context.DetailedErr,
+			Migrated: &context.Migrated,
+		}
+		c.Err = &detailedErr
```

Comment:
do we need this line?

Reply:
Nice catch! I will delete this in another PR